src/config.rs

```diff
@@ -30,12 +47,26 @@ impl Cfg {
             return Err(miette::miette!("data_dir cannot be empty"));
         }
 
-        if cfg.storage_compression.is_some_and(|v| v) {
-            println!("Storage compression is enabled");
-        }
-
-        if cfg.output_compression.is_some_and(|v| v) {
-            println!("Output compression is enabled");
+        if let Some(ref comp) = cfg.storage_compression {
+            if comp.enabled {
+                
+                let level = comp.level.unwrap();
+                let valid = match comp.algorithm {
+                    CompressionType::None | CompressionType::Snappy => false,
+                    CompressionType::Gzip => level >= 1 && level <= 9,
+                    CompressionType::Zstd => level >= 1 && level <= 22,
+                    CompressionType::Lz4 => level >= 1 && level <= 16,
```

Comment:
Instead of this, have an initialization function in the compression crate, which will take cfg and init the correct compression impl